src/tcp_server.rs

```diff
@@ -17,12 +14,38 @@ use crate::ServerMockerError::{
     UnableToWriteTcpStream,
 };
 
+// FIXME: consider consolidating options for both TCP & UDP
+/// Options for the TCP server mocker
+#[derive(Debug, Clone)]
+pub struct TcpMockerOptions {
+    /// Socket address on which the server will listen. Will be set to `127.0.0.1:0` by default.
+    pub socket_addr: SocketAddr,
+    /// Timeout for the server to wait for a message from the client.
+    pub net_timeout: Duration,
+    /// Timeout if no more instruction is available and [`Instruction::StopExchange`] hasn't been sent
+    pub rx_timeout: Duration,
```

Comment:
I think `rx_timeout` isn't very explicit, agree?

Reply:
I do agree that it is not ideal - I was trying to come up with a concise and descriptive name for this, and I suspect we will have to rework all of the timeout concepts anyway:  we need "first byte received", vs "timeout between bytes in one message", vs possibly other... maybe.  Plus naming should be consistent with how they are named in the std lib.  `rx` is a common abbreviation -- see [here](https://doc.rust-lang.org/std/sync/mpsc/struct.Sender.html) -- it uses both `rx` and `receiver`.